clang/test/CodeGen/cheri/subobject-bounds-pointer-container.c

```diff
@@ -0,0 +1,52 @@
+// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py
+// RUN: %cheri_purecap_cc1 -cheri-bounds=subobject-safe -disable-O0-optnone -emit-llvm %s -o - \
```

Comment:
Could you add `-Rcheri-subobject-bounds -verify` to this test?

The https://cheri-compiler-explorer.cl.cam.ac.uk/z/64ofor output shows that it tries to use the union size:
```
remark: using size of containing type 'union (anonymous union at <source>:2:3)' instead of object type 'char' for subobject bounds on union member [-Rcheri-subobject-bounds]
  return &s->u->p[1];
```